### PR TITLE
bpo-5755: remove compiler flag -Wstrict-prototypes

### PR DESCRIPTION
**Pull request by jdemeyer**
https://bugs.python.org/issue5755

### CODE REVIEW

**Comment by jdemeyer** [comment]
CCC @grimreaper

**Comment by grimreaper** [comment]
Strict Prototypes is correct and helpful for C. That said, the way it is currently implemented, by checking for the compiler in CC is sub-optimal at best.
CC ought to always be a C compiler, and never a C++ compiler. 
It also seems questionable that distutils is using the python cflags rather than a dedicated set.

Please also remove ` 	OPT="$OPT $STRICT_PROTO"` and any other reference to STRICT_PROTO since it is now unused.

TL;DR; +1 to this change; distutils should be fixed to not use python's CFLAGS; we should re-add strict-prototypes to CFLAGS when distutils is fixed.

**Review by methane — APPROVED** [review]

**Comment by methane** [comment]
PR-7395 was merged, and backported to 3.7 and 3.6.
